arroyo-sql/src/lib.rs

```diff
@@ -70,7 +72,7 @@ pub struct UdfDef {
 #[derive(Debug, Clone, Default)]
 pub struct ArroyoSchemaProvider {
     pub source_defs: HashMap<String, String>,
-    tables: HashMap<String, Table>,
+    tables: HashMap<UniCase<String>, Table>,
```

Comment:
This seems good, given the case insensitive nature of SQL. Can you add a check to when connections are created, or automatically lowercase them?

Reply:
I've added a new index to the connections_table table that prevents creating multiple tables with the same (case-insensitive) name